diff -r -U3 kig-22.12.0.orig/CMakeLists.txt kig-22.12.0/CMakeLists.txt
--- kig-22.12.0.orig/CMakeLists.txt	2020-08-07 09:54:58.000000000 +0300
+++ kig-22.12.0/CMakeLists.txt	2020-08-16 23:18:22.463314427 +0300
@@ -72,6 +72,9 @@
 if(BoostPython_FOUND)
   add_subdirectory( scripting )
   add_definitions(-DKIG_ENABLE_PYTHON_SCRIPTING)
+  get_filename_component(PYTHON_LIBRARY_REALPATH "${PYTHON_LIBRARY}" REALPATH)
+  get_filename_component(LIB_PYTHON ${PYTHON_LIBRARY_REALPATH} NAME)
+  add_definitions(-DLIB_PYTHON="${LIB_PYTHON}")
 endif(BoostPython_FOUND)
 
 set_package_properties(
diff -r -U3 kig-22.12.0.orig/scripting/python_scripter.cc kig-22.12.0/scripting/python_scripter.cc
--- kig-22.12.0.orig/scripting/python_scripter.cc	2020-08-07 08:49:45.000000000 +0300
+++ kig-22.12.0/scripting/python_scripter.cc	2020-08-16 23:18:22.463314427 +0300
@@ -10,6 +10,8 @@
 #include "python_scripter.h"
 #include <Python.h>
 
+#include <QLibrary>
+
 #include <iostream>
 #include <string>
 
@@ -311,6 +311,11 @@
 
     PyImport_AppendInittab("kig", PyInit_kig);
 
+    QLibrary *pythonLib = new QLibrary();
+    pythonLib->setLoadHints(QLibrary::ExportExternalSymbolsHint);
+    pythonLib->setFileName(LIB_PYTHON);
+    pythonLib->load();
+
     Py_Initialize();
 
     PyRun_SimpleString("import math; from math import *;");
