From adf53dd4bdfdbf98b2b62061c586d4c078aa9c41 Mon Sep 17 00:00:00 2001
From: Yaroslav Sidlovsky <zawertun@gmail.com>
Date: Wed, 6 Jan 2021 01:34:43 +0300
Subject: [PATCH] Don't copy empty exporteddata information file

BUG: 429895
---
 core/pimdataimportdatainfofile.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/core/pimdataimportdatainfofile.cpp b/core/pimdataimportdatainfofile.cpp
index b1311dd6..de106f95 100644
--- a/core/pimdataimportdatainfofile.cpp
+++ b/core/pimdataimportdatainfofile.cpp
@@ -40,11 +40,15 @@ QString PimDataImportDataInfoFile::importDataInfoPath()
         const KArchiveEntry *informationFile = archive->directory()->entry(Utils::infoPath() + Utils::exportDataTypeFileName());
         if (informationFile && informationFile->isFile()) {
             const auto *file = static_cast<const KArchiveFile *>(informationFile);
-            temporaryFilePath = mTempDir->path() + QLatin1Char('/') + Utils::exportDataTypeFileName();
-            if (file->copyTo(mTempDir->path())) {
-                temporaryFilePath = mTempDir->path() + QLatin1Char('/') + Utils::exportDataTypeFileName();
+            if (file->size() == 0) {
+                qCWarning(PIMDATAEXPORTERCORE_LOG) << "Empty exporteddata information file, skipping it";
             } else {
-                qCWarning(PIMDATAEXPORTERCORE_LOG) << "Impossible to copy to temporary file" << temporaryFilePath;
+                temporaryFilePath = mTempDir->path() + QLatin1Char('/') + Utils::exportDataTypeFileName();
+                if (file->copyTo(mTempDir->path())) {
+                    temporaryFilePath = mTempDir->path() + QLatin1Char('/') + Utils::exportDataTypeFileName();
+                } else {
+                    qCWarning(PIMDATAEXPORTERCORE_LOG) << "Impossible to copy to temporary file" << temporaryFilePath;
+                }
             }
         } else {
             qCWarning(PIMDATAEXPORTERCORE_LOG) << "Old archive without exporteddata information";
-- 
GitLab

