diff -up konqueror-18.12.2/CMakeLists.txt.opt konqueror-18.12.2/CMakeLists.txt
--- konqueror-18.12.2/CMakeLists.txt.opt	2019-02-04 18:14:27.000000000 -0600
+++ konqueror-18.12.2/CMakeLists.txt	2019-02-20 12:07:12.717586668 -0600
@@ -23,7 +23,8 @@ include(ECMQtDeclareLoggingCategory)
 set(KONQUEROR_LIB_VERSION "5.0.97")
 set(KONQUEROR_VERSION "${KONQUEROR_LIB_VERSION}")
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets WebEngineWidgets)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets)
+find_package(Qt5WebEngineWidgets)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Parts KCMUtils KHtml KDELibs4Support Archive Crash)
 
 find_package(KF5 ${KF5_MIN_VERSION} COMPONENTS Activities DocTools) # Optional
@@ -55,7 +56,9 @@ add_subdirectory( client )
 if(BUILD_TESTING)
     add_subdirectory( autotests )
 endif()
+if (Qt5WebEngineWidgets_FOUND)
 add_subdirectory( webenginepart )
+endif()
 
 add_subdirectory( about )
 add_subdirectory( pics )
