diff -r -U3 kdenlive-19.04.0.orig/src/mltconnection.cpp kdenlive-19.04.0/src/mltconnection.cpp
--- kdenlive-19.04.0.orig/src/mltconnection.cpp	2019-04-29 22:21:25.203806292 +0300
+++ kdenlive-19.04.0/src/mltconnection.cpp	2019-04-29 22:21:36.485849112 +0300
@@ -101,9 +101,9 @@
     if (!QFile::exists(meltPath)) {
         meltPath = QDir::cleanPath(profilePath + QStringLiteral("/../../../bin/melt")) + exeSuffix;
         if (!QFile::exists(meltPath)) {
-            meltPath = QStandardPaths::findExecutable("melt");
+            meltPath = QStandardPaths::findExecutable("mlt-melt");
             if (meltPath.isEmpty()) {
-                meltPath = QStandardPaths::findExecutable("mlt-melt");
+                meltPath = QStandardPaths::findExecutable("melt");
             }
         }
     }
