diff --git a/src/ioslaves/ftp/ftp.cpp b/src/ioslaves/ftp/ftp.cpp
index 2c07ceb9..744e3b39 100644
--- a/src/ioslaves/ftp/ftp.cpp
+++ b/src/ioslaves/ftp/ftp.cpp
@@ -2372,7 +2372,7 @@ void Ftp::copy(const QUrl &src, const QUrl &dest, int permissions, KIO::JobFlags
         QT_CLOSE(iCopyFile);
     }
     ftpCloseCommand();                        // must close command!
-    if (cs == statusServerError && iError) {
+    if (iError) {
         error(iError, sCopyFile);
     } else {
         finished();
