diff -up kio-5.40.0/src/core/slave.cpp.path kio-5.40.0/src/core/slave.cpp
--- kio-5.40.0/src/core/slave.cpp.path	2017-11-05 02:33:04.000000000 -0600
+++ kio-5.40.0/src/core/slave.cpp	2017-11-13 10:04:14.921095694 -0600
@@ -504,7 +504,6 @@ Slave *Slave::createSlave(const QString
 
         // search paths
         const QStringList searchPaths = QStringList()
-            << QCoreApplication::applicationDirPath() // then look where our application binary is located
             << QLibraryInfo::location(QLibraryInfo::LibraryExecutablesPath) // look where libexec path is (can be set in qt.conf)
             << QFile::decodeName(CMAKE_INSTALL_FULL_LIBEXECDIR_KF5); // look at our installation location
         const QString kioslaveExecutable = QStandardPaths::findExecutable(QStringLiteral("kioslave"), searchPaths);
