diff -up kio-extras-19.04.2/CMakeLists.txt.nfs kio-extras-19.04.2/CMakeLists.txt
--- kio-extras-19.04.2/CMakeLists.txt.nfs	2019-06-03 19:13:48.000000000 -0500
+++ kio-extras-19.04.2/CMakeLists.txt	2019-06-05 14:02:00.219046753 -0500
@@ -178,9 +178,11 @@
    if(Gperf_FOUND)
      add_subdirectory( man )
    endif()
-   check_include_files(rpc/rpc.h HAVE_RPC_RPC_H)
-   add_feature_info("NFS kioslave" HAVE_RPC_RPC_H "The RPC library is needed to build the NFS kioslave")
-   if(HAVE_RPC_RPC_H)
+   pkg_check_modules(TIRPC libtirpc)
+   include_directories(${TIRPC_INCLUDE_DIRS})
+   set(CMAKE_REQUIRED_LIBRARIES ${TIRPC_LIBRARIES})
+   add_feature_info("NFS kioslave" TIRPC_FOUND "The RPC library is needed to build the NFS kioslave")
+   if(TIRPC_FOUND)
      add_subdirectory( nfs )
    endif()
 endif()
diff -up kio-extras-19.04.2/nfs/CMakeLists.txt.nfs kio-extras-19.04.2/nfs/CMakeLists.txt
--- kio-extras-19.04.2/nfs/CMakeLists.txt.nfs	2019-05-31 05:33:44.000000000 -0500
+++ kio-extras-19.04.2/nfs/CMakeLists.txt	2019-06-05 14:02:33.321631526 -0500
@@ -10,6 +10,7 @@ if (NOT HAVE_XDR_U_INT64_T AND NOT HAVE_
     message(FATAL_ERROR "Could not find 64-bit XDR datatype functions!")
 endif (NOT HAVE_XDR_U_INT64_T AND NOT HAVE_XDR_UINT64_T AND NOT HAVE_XDR_U_HYPER AND NOT HAVE_XDR_U_LONGLONG_T)
 
+add_definitions(-D_DEFAULT_SOURCE)
 if (HAVE_XDR_U_INT64_T)
     add_definitions(-DHAVE_XDR_U_INT64_T)
 endif (HAVE_XDR_U_INT64_T)
@@ -26,7 +27,8 @@ endif (HAVE_XDR_U_LONGLONG_T)
 add_definitions(-DTRANSLATION_DOMAIN=\"kio5_nfs\")
 
 add_library(kio_nfs MODULE kio_nfs.cpp nfsv2.cpp nfsv3.cpp rpc_nfs3_prot_xdr.c rpc_nfs2_prot_xdr.c)
-target_link_libraries(kio_nfs KF5::KIOCore KF5::I18n Qt5::Network)
+target_link_libraries(kio_nfs KF5::KIOCore KF5::I18n Qt5::Network ${TIRPC_LIBRARIES})
+
 set_target_properties(kio_nfs PROPERTIES OUTPUT_NAME "nfs")
 
 install(TARGETS kio_nfs DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/kio)
