From c326b72cce5eb7a72c32de9458127bd38e553cf8 Mon Sep 17 00:00:00 2001
From: Yaroslav Sidlovsky <zawertun@gmail.com>
Date: Tue, 14 Sep 2021 20:15:01 +0300
Subject: [PATCH] Fix opening RAW files with embedded TIFF previews

BUG: 441698
---
 lib/document/loadingdocumentimpl.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/document/loadingdocumentimpl.cpp b/lib/document/loadingdocumentimpl.cpp
index 8471bbd0..8503fde9 100644
--- a/lib/document/loadingdocumentimpl.cpp
+++ b/lib/document/loadingdocumentimpl.cpp
@@ -121,6 +121,12 @@ struct LoadingDocumentImplPrivate {
             mime = db.mimeTypeForFileNameAndData(url.fileName(), mData);
         }
 
+        // Some RAW files has embedded TIFF previews => detection by content will fail
+        auto fileInfo = QFileInfo(url.fileName());
+        if (mime.name() == "image/tiff" && !mime.suffixes().contains(fileInfo.suffix().toLower())) {
+            mime = db.mimeTypeForFileNameAndData(url.fileName(), mData);
+        }
+
         mMimeType = mime;
 
         MimeTypeUtils::Kind kind = MimeTypeUtils::mimeTypeKind(mMimeType.name());
-- 
GitLab

