diff --git a/krusader/FileSystem/filesystem.cpp b/krusader/FileSystem/filesystem.cpp
index c05c47f5..17147cfd 100644
--- a/krusader/FileSystem/filesystem.cpp
+++ b/krusader/FileSystem/filesystem.cpp
@@ -94,10 +94,12 @@ bool FileSystem::scanOrRefresh(const QUrl &directory, bool onlyScan)
 
     // workaround for krarc: find out if transition to local fs is wanted and adjust URL manually
     QUrl url = directory;
+#if KIO_VERSION < QT_VERSION_CHECK(5, 95, 0)
     if (_currentDirectory.scheme() == "krarc" && url.scheme() == "krarc" &&
         QDir(url.path()).exists()) {
         url.setScheme("file");
     }
+#endif
 
     const bool dirChange = !url.isEmpty() && cleanUrl(url) != _currentDirectory;
 
