diff --git a/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp b/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
index 6e2c5c232e6642a4ec9e0f39c90bf0a5971b97eb..7670a7c189d8fc9c20a0ae1d87e0c84317ec9c2d 100644
--- a/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
+++ b/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
@@ -18,6 +18,8 @@
 #include <KConfigGroup>
 #include <QDBusConnection>
 
+#include <cmath>
+
 class IconLoaderSingleton
 {
 public:
@@ -54,6 +56,14 @@ public:
         QStringLiteral( "/KGlobalSettings" ),
         QStringLiteral( "org.kde.KGlobalSettings" ),
         QStringLiteral( "notifyChange" ), this, SIGNAL(configurationChanged()));
+
+        // Use NativeTextRendering as the default text rendering type when the scale factor is an integer.
+        // NativeTextRendering is still distorted sometimes with fractional scale factors,
+        // despite https://bugreports.qt.io/browse/QTBUG-67007 being closed.
+        qreal devicePixelRatio = qGuiApp->devicePixelRatio();
+        QQuickWindow::TextRenderType defaultTextRenderType = int(devicePixelRatio) == devicePixelRatio ?
+            QQuickWindow::NativeTextRendering : QQuickWindow::QtTextRendering;
+        QQuickWindow::setTextRenderType(defaultTextRenderType);
     }
 
     void refresh()
