diff -up kpimtextedit-20.04.0/src/CMakeLists.txt.install_and_export_for_blogilo kpimtextedit-20.04.0/src/CMakeLists.txt
--- kpimtextedit-20.04.0/src/CMakeLists.txt.install_and_export_for_blogilo	2020-04-15 06:33:58.000000000 -0500
+++ kpimtextedit-20.04.0/src/CMakeLists.txt	2020-04-24 09:21:37.533930897 -0500
@@ -114,6 +114,9 @@ install(TARGETS KF5PimTextEdit EXPORT KF
 ecm_generate_headers(KPimTextEdit_CamelCase_HEADERS
     HEADER_NAMES
     EditorUtil
+    InsertHtmlDialog
+    InsertImageWidget
+    InsertTableWidget
     SelectSpecialCharDialog
     SlideContainer
     TextEditorCompleter
@@ -125,6 +128,7 @@ ecm_generate_headers(KPimTextEdit_CamelC
 
 ecm_generate_headers(KPimTextEdit_CamelCaseemoticon_HEADERS
     HEADER_NAMES
+    EmoticonTextEditAction
     EmoticonUnicodeTab
     PREFIX KPIMTextEdit
     REQUIRED_HEADERS kpimtextedit_HEADERS
diff -up kpimtextedit-20.04.0/src/emoticon/emoticontexteditaction.h.install_and_export_for_blogilo kpimtextedit-20.04.0/src/emoticon/emoticontexteditaction.h
--- kpimtextedit-20.04.0/src/emoticon/emoticontexteditaction.h.install_and_export_for_blogilo	2020-04-15 06:33:58.000000000 -0500
+++ kpimtextedit-20.04.0/src/emoticon/emoticontexteditaction.h	2020-04-24 09:20:46.232419048 -0500
@@ -14,7 +14,7 @@
 
 namespace KPIMTextEdit
 {
-class EmoticonTextEditAction : public KActionMenu
+class KPIMTEXTEDIT_EXPORT EmoticonTextEditAction : public KActionMenu
 {
     Q_OBJECT
 public:
diff -up kpimtextedit-20.04.0/src/inserthtmldialog.h.install_and_export_for_blogilo kpimtextedit-20.04.0/src/inserthtmldialog.h
--- kpimtextedit-20.04.0/src/inserthtmldialog.h.install_and_export_for_blogilo	2020-04-15 06:33:58.000000000 -0500
+++ kpimtextedit-20.04.0/src/inserthtmldialog.h	2020-04-24 09:20:46.232419048 -0500
@@ -11,12 +11,12 @@
 #include <QDialog>
 
 #include "kpimtextedit/plaintexteditor.h"
-#include "kpimtextedit_private_export.h"
+#include "kpimtextedit_export.h"
 
 namespace KPIMTextEdit
 {
 class InsertHtmlDialogPrivate;
-class KPIMTEXTEDIT_TESTS_EXPORT InsertHtmlDialog : public QDialog
+class KPIMTEXTEDIT_EXPORT InsertHtmlDialog : public QDialog
 {
     Q_OBJECT
 public:
diff -up kpimtextedit-20.04.0/src/insertimagewidget.h.install_and_export_for_blogilo kpimtextedit-20.04.0/src/insertimagewidget.h
--- kpimtextedit-20.04.0/src/insertimagewidget.h.install_and_export_for_blogilo	2020-04-15 06:33:58.000000000 -0500
+++ kpimtextedit-20.04.0/src/insertimagewidget.h	2020-04-24 09:20:46.232419048 -0500
@@ -7,6 +7,8 @@
 #ifndef KPIMTEXTEDIT_INSERTIMAGEWIDGET_H
 #define KPIMTEXTEDIT_INSERTIMAGEWIDGET_H
 
+#include "kpimtextedit_export.h"
+
 #include <QUrl>
 
 #include <QWidget>
@@ -15,7 +17,7 @@
 {
 class InsertImageWidgetPrivate;
 
-class InsertImageWidget : public QWidget
+class KPIMTEXTEDIT_EXPORT InsertImageWidget : public QWidget
 {
     Q_OBJECT
 public:
diff -up kpimtextedit-20.04.0/src/inserttablewidget.h.install_and_export_for_blogilo kpimtextedit-20.04.0/src/inserttablewidget.h
--- kpimtextedit-20.04.0/src/inserttablewidget.h.install_and_export_for_blogilo	2020-04-15 06:33:58.000000000 -0500
+++ kpimtextedit-20.04.0/src/inserttablewidget.h	2020-04-24 09:20:46.232419048 -0500
@@ -7,6 +7,7 @@
 
 #ifndef INSERTTABLEWIDGET_H
 #define INSERTTABLEWIDGET_H
+#include "kpimtextedit_export.h"
 
 #include <QWidget>
 
@@ -14,7 +15,7 @@
 
 namespace KPIMTextEdit
 {
-class InsertTableWidget : public QWidget
+class KPIMTEXTEDIT_EXPORT InsertTableWidget : public QWidget
 {
     Q_OBJECT
 public:
