diff --git a/src/kkeysequencewidget.cpp b/src/kkeysequencewidget.cpp
index a51c6cc22a37c3a74e726cfa70dafe02830183da..c1f5c42c506ecf27e508442690700eac2bf00194 100644
--- a/src/kkeysequencewidget.cpp
+++ b/src/kkeysequencewidget.cpp
@@ -493,6 +493,7 @@ void KKeySequenceWidget::setCheckActionCollections(const QList<KActionCollection
 //slot
 void KKeySequenceWidget::captureKeySequence()
 {
+    d->recorder->setWindow(window()->windowHandle());
     d->recorder->startRecording();
 }
 
