diff -up ksystemlog-15.12.1/src/modes/apache/apacheConfiguration.h.fedora ksystemlog-15.12.1/src/modes/apache/apacheConfiguration.h
--- ksystemlog-15.12.1/src/modes/apache/apacheConfiguration.h.fedora	2015-11-15 06:46:00.000000000 -0600
+++ ksystemlog-15.12.1/src/modes/apache/apacheConfiguration.h	2016-01-24 08:10:24.555020724 -0600
@@ -52,12 +52,12 @@ public:
         configuration->setCurrentGroup(QStringLiteral("ApacheLogMode"));
 
         QStringList defaultApachePaths;
-        defaultApachePaths << QStringLiteral("/var/log/apache2/error.log");
+        defaultApachePaths << QStringLiteral("/var/log/httpd/error.log");
         configuration->addItemStringList(QStringLiteral("ApacheLogFilesPaths"), d->apachePaths,
                                          defaultApachePaths, QStringLiteral("ApacheLogFilesPaths"));
 
         QStringList defaultApacheAccessPaths;
-        defaultApacheAccessPaths << QStringLiteral("/var/log/apache2/access.log");
+        defaultApacheAccessPaths << QStringLiteral("/var/log/httpd/access.log");
         configuration->addItemStringList(QStringLiteral("ApacheAccessLogFilesPaths"), d->apacheAccessPaths,
                                          defaultApacheAccessPaths,
                                          QStringLiteral("ApacheAccessLogFilesPaths"));
diff -up ksystemlog-15.12.1/src/modes/authentication/authenticationConfiguration.h.fedora ksystemlog-15.12.1/src/modes/authentication/authenticationConfiguration.h
--- ksystemlog-15.12.1/src/modes/authentication/authenticationConfiguration.h.fedora	2015-11-15 06:46:00.000000000 -0600
+++ ksystemlog-15.12.1/src/modes/authentication/authenticationConfiguration.h	2016-01-24 08:09:27.958487885 -0600
@@ -52,7 +52,7 @@ public:
     {
         configuration->setCurrentGroup(QStringLiteral("AuthenticationLogMode"));
 
-        QString defaultAuthenticationPath(QStringLiteral("/var/log/auth.log"));
+        QString defaultAuthenticationPath(QStringLiteral("/var/log/secure"));
         configuration->addItemString(QStringLiteral("LogFilePath"), d->authenticationPath,
                                      defaultAuthenticationPath, QStringLiteral("LogFilePath"));
 
diff -up ksystemlog-15.12.1/src/modes/cron/cronConfiguration.h.fedora ksystemlog-15.12.1/src/modes/cron/cronConfiguration.h
--- ksystemlog-15.12.1/src/modes/cron/cronConfiguration.h.fedora	2015-11-15 06:46:00.000000000 -0600
+++ ksystemlog-15.12.1/src/modes/cron/cronConfiguration.h	2016-01-24 08:11:07.992429675 -0600
@@ -52,7 +52,7 @@ public:
         configuration->setCurrentGroup(QStringLiteral("CronLogMode"));
 
         QStringList defaultCronPaths;
-        defaultCronPaths << QStringLiteral("/var/log/syslog");
+        defaultCronPaths << QStringLiteral("/var/log/cron");
         configuration->addItemStringList(QStringLiteral("LogFilesPaths"), d->cronPaths, defaultCronPaths,
                                          QStringLiteral("LogFilesPaths"));
 
diff -up ksystemlog-15.12.1/src/modes/cups/cupsConfiguration.h.fedora ksystemlog-15.12.1/src/modes/cups/cupsConfiguration.h
diff -up ksystemlog-15.12.1/src/modes/daemon/daemonConfiguration.h.fedora ksystemlog-15.12.1/src/modes/daemon/daemonConfiguration.h
--- ksystemlog-15.12.1/src/modes/daemon/daemonConfiguration.h.fedora	2015-11-15 06:46:00.000000000 -0600
+++ ksystemlog-15.12.1/src/modes/daemon/daemonConfiguration.h	2016-01-24 08:13:12.500601882 -0600
@@ -48,7 +48,7 @@ public:
         configuration->setCurrentGroup(QStringLiteral("DaemonLogMode"));
 
         QStringList defaultDaemonPaths;
-        defaultDaemonPaths << QStringLiteral("/var/log/daemon.log");
+        defaultDaemonPaths << QStringLiteral("/var/log/boot.log");
         configuration->addItemStringList(QStringLiteral("LogFilesPaths"), d->daemonPaths, defaultDaemonPaths,
                                          QStringLiteral("LogFilesPaths"));
     }
diff -up ksystemlog-15.12.1/src/modes/postfix/postfixConfiguration.h.fedora ksystemlog-15.12.1/src/modes/postfix/postfixConfiguration.h
--- ksystemlog-15.12.1/src/modes/postfix/postfixConfiguration.h.fedora	2016-01-24 08:13:59.820047381 -0600
+++ ksystemlog-15.12.1/src/modes/postfix/postfixConfiguration.h	2016-01-24 08:14:28.905321211 -0600
@@ -39,8 +39,7 @@ public:
     PostfixConfiguration()
         : GenericLogModeConfiguration(
               QStringLiteral(POSTFIX_LOG_MODE_ID),
-              QStringList() << QStringLiteral("/var/log/mail.log") << QStringLiteral("/var/log/mail.info")
-                            << QStringLiteral("/var/log/mail.warn") << QStringLiteral("/var/log/mail.err"),
+              QStringList() << QStringLiteral("/var/log/maillog"),
               QList<int>() << Globals::NOTICE_LOG_LEVEL_ID << Globals::INFORMATION_LOG_LEVEL_ID << Globals::WARNING_LOG_LEVEL_ID
                            << Globals::ERROR_LOG_LEVEL_ID)
     {
diff -up ksystemlog-15.12.1/src/modes/system/systemConfiguration.h.fedora ksystemlog-15.12.1/src/modes/system/systemConfiguration.h
--- ksystemlog-15.12.1/src/modes/system/systemConfiguration.h.fedora	2015-11-15 06:46:00.000000000 -0600
+++ ksystemlog-15.12.1/src/modes/system/systemConfiguration.h	2016-01-24 08:15:58.863168138 -0600
@@ -38,7 +38,7 @@ class SystemConfiguration : public Gener
 public:
     SystemConfiguration()
         : GenericLogModeConfiguration(QStringLiteral(SYSTEM_LOG_MODE_ID),
-                                      QStringList() << QStringLiteral("/var/log/syslog"),
+                                      QStringList() << QStringLiteral("/var/log/messages"),
                                       QList<int>() << Globals::INFORMATION_LOG_LEVEL_ID)
     {
         /*
