diff -up ktp-text-ui-15.04.2/image-sharer/CMakeLists.txt.soversion ktp-text-ui-15.04.2/image-sharer/CMakeLists.txt
--- ktp-text-ui-15.04.2/image-sharer/CMakeLists.txt.soversion	2015-05-28 17:03:46.000000000 -0500
+++ ktp-text-ui-15.04.2/image-sharer/CMakeLists.txt	2015-06-10 07:03:18.122890799 -0500
@@ -13,5 +13,8 @@ target_link_libraries(ktpimagesharer
                       KF5::KIOWidgets
                       KTp::OTR
 )
+set_target_properties(ktpimagesharer PROPERTIES
+                      SOVERSION ${KTP_TEXT_UI_VERSION}
+)
 
 install(TARGETS ktpimagesharer ${INSTALL_TARGETS_DEFAULT_ARGS})
diff -up ktp-text-ui-15.04.2/lib/CMakeLists.txt.soversion ktp-text-ui-15.04.2/lib/CMakeLists.txt
--- ktp-text-ui-15.04.2/lib/CMakeLists.txt.soversion	2015-06-10 07:03:18.122890799 -0500
+++ ktp-text-ui-15.04.2/lib/CMakeLists.txt	2015-06-10 07:04:07.198982480 -0500
@@ -33,6 +33,9 @@ set(ktpchat_UI
 ki18n_wrap_ui(ktpchat_SRCS ${ktpchat_UI})
 
 add_library(ktpchat SHARED ${ktpchat_SRCS})
+set_target_properties(ktpchat PROPERTIES
+    SOVERSION ${KTP_TEXT_UI_VERSION}
+)
 
 target_link_libraries(ktpchat
     KF5::KIOWidgets
