diff --git a/processui/ReniceDlg.cpp b/processui/ReniceDlg.cpp
index b8fa4be..70329ed 100644
--- a/processui/ReniceDlg.cpp
+++ b/processui/ReniceDlg.cpp
@@ -188,8 +188,8 @@ void ReniceDlg::setSliderRange() {
 		//The slider is setting the priority, so goes from 19 to -20.  We cannot actually do this with a slider, so instead we go from -19 to 20, and negate later
 		if(ui->sliderCPU->value() > 20) ui->sliderCPU->setValue(20);
 		ui->sliderCPU->setInvertedAppearance(true);
-		ui->sliderCPU->setMinimum(-19);
-		ui->sliderCPU->setMaximum(20);
+		ui->sliderCPU->setMinimum(-20);
+		ui->sliderCPU->setMaximum(19);
 		ui->sliderCPU->setTickInterval(5);
 	} else {
 		if(ui->sliderCPU->value() < 1) ui->sliderCPU->setValue(1);
diff --git a/processui/ReniceDlgUi.ui b/processui/ReniceDlgUi.ui
index 02ef7ee..e3e34da 100644
--- a/processui/ReniceDlgUi.ui
+++ b/processui/ReniceDlgUi.ui
@@ -371,10 +371,10 @@ p, li { white-space: pre-wrap; }
        <item row="0" column="0" colspan="2">
         <widget class="QSlider" name="sliderCPU">
          <property name="minimum">
-          <number>-19</number>
+          <number>-20</number>
          </property>
          <property name="maximum">
-          <number>20</number>
+          <number>19</number>
          </property>
          <property name="value">
           <number>0</number>
