diff -up digikam-5.9.0/doc-translated/ca/CMakeLists.txt.doc_translated digikam-5.9.0/doc-translated/ca/CMakeLists.txt
--- digikam-5.9.0/doc-translated/ca/CMakeLists.txt.doc_translated	2018-03-19 04:42:01.000000000 -0500
+++ digikam-5.9.0/doc-translated/ca/CMakeLists.txt	2018-03-28 07:44:44.626219821 -0500
@@ -1,2 +1,2 @@
 KDOCTOOLS_CREATE_HANDBOOK( digikam/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/ca/ SUBDIR digikam )
-KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/ca/ SUBDIR digikam )
\ No newline at end of file
+KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/ca/ SUBDIR showfoto )
diff -up digikam-5.9.0/doc-translated/nl/CMakeLists.txt.doc_translated digikam-5.9.0/doc-translated/nl/CMakeLists.txt
--- digikam-5.9.0/doc-translated/nl/CMakeLists.txt.doc_translated	2018-03-19 04:42:01.000000000 -0500
+++ digikam-5.9.0/doc-translated/nl/CMakeLists.txt	2018-03-28 07:42:57.451481343 -0500
@@ -1,2 +1,2 @@
 KDOCTOOLS_CREATE_HANDBOOK( digikam/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/nl/ SUBDIR digikam )
-KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/nl/ SUBDIR digikam )
\ No newline at end of file
+KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/nl/ SUBDIR showfoto )
diff -up digikam-5.9.0/doc-translated/sv/CMakeLists.txt.doc_translated digikam-5.9.0/doc-translated/sv/CMakeLists.txt
--- digikam-5.9.0/doc-translated/sv/CMakeLists.txt.doc_translated	2018-03-19 04:42:01.000000000 -0500
+++ digikam-5.9.0/doc-translated/sv/CMakeLists.txt	2018-03-28 07:42:57.453481357 -0500
@@ -1,2 +1,2 @@
 KDOCTOOLS_CREATE_HANDBOOK( digikam/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/sv/ SUBDIR digikam )
-KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/sv/ SUBDIR digikam )
\ No newline at end of file
+KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/sv/ SUBDIR showfoto )
diff -up digikam-5.9.0/doc-translated/uk/CMakeLists.txt.doc_translated digikam-5.9.0/doc-translated/uk/CMakeLists.txt
--- digikam-5.9.0/doc-translated/uk/CMakeLists.txt.doc_translated	2018-03-19 04:42:01.000000000 -0500
+++ digikam-5.9.0/doc-translated/uk/CMakeLists.txt	2018-03-28 07:42:57.453481357 -0500
@@ -1,2 +1,2 @@
 KDOCTOOLS_CREATE_HANDBOOK( digikam/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/uk/ SUBDIR digikam )
-KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/uk/ SUBDIR digikam )
\ No newline at end of file
+KDOCTOOLS_CREATE_HANDBOOK( showfoto/index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/uk/ SUBDIR showfoto )
