diff -up digikam-6.2.0/core/libs/facesengine/CMakeLists.txt.Wall digikam-6.2.0/core/libs/facesengine/CMakeLists.txt
--- digikam-6.2.0/core/libs/facesengine/CMakeLists.txt.Wall	2019-07-31 11:12:57.000000000 -0500
+++ digikam-6.2.0/core/libs/facesengine/CMakeLists.txt	2019-08-03 09:56:28.694088905 -0500
@@ -95,7 +95,7 @@ if(ENABLE_FACESENGINE_DNN)
         # Dlib DNN generate plenty of wrong warnings. Remove these workaround later when code will be ported to OpenCV DNN.
         REMOVE_GCC_COMPILER_WARNINGS("-Wextra")
         REMOVE_GCC_COMPILER_WARNINGS("-Wundef")
-        REMOVE_GCC_COMPILER_WARNINGS("-Wall")
+        #REMOVE_GCC_COMPILER_WARNINGS("-Wall")
 
         DISABLE_CLANG_COMPILER_WARNINGS("4.99.99" "-Wno-cast-align")
         DISABLE_CLANG_COMPILER_WARNINGS("4.99.99" "-Wno-sign-compare")
