diff -up kiconthemes-5.60.0/src/kicontheme.cpp.logspam kiconthemes-5.60.0/src/kicontheme.cpp
--- kiconthemes-5.60.0/src/kicontheme.cpp.logspam	2019-07-07 13:34:23.000000000 -0500
+++ kiconthemes-5.60.0/src/kicontheme.cpp	2019-07-18 08:25:14.857727321 -0500
@@ -764,7 +764,7 @@ KIconThemeDir::KIconThemeDir(const QStri
     } else if (tmp.isEmpty()) {
         // do nothing. key not required
     } else {
-        qWarning() << "Invalid Context=" << tmp << "line for icon theme: " << constructFileName(QString());
+        qDebug() << "Invalid Context=" << tmp << "line for icon theme: " << constructFileName(QString());
         return;
     }
     tmp = config.readEntry(QStringLiteral("Type"), QStringLiteral("Threshold"));
