diff -up kdenlive-18.12.3/src/mltconnection.cpp.mlt kdenlive-18.12.3/src/mltconnection.cpp
--- kdenlive-18.12.3/src/mltconnection.cpp.mlt	2019-02-27 19:43:02.000000000 -0600
+++ kdenlive-18.12.3/src/mltconnection.cpp	2019-03-29 08:34:40.715613792 -0500
@@ -73,7 +73,10 @@ void MltConnection::locateMeltAndProfile
         meltPath = QDir::cleanPath(profilePath + QStringLiteral("/../../../bin/melt")) + exeSuffix;
     }
     if (!QFile::exists(meltPath)) {
-        meltPath = QStandardPaths::findExecutable("melt");
+        meltPath = QStandardPaths::findExecutable("mlt-melt");
+        if (meltPath.isEmpty()) {
+            meltPath = QStandardPaths::findExecutable("melt");
+        }
     }
     KdenliveSettings::setRendererpath(meltPath);
 
