diff -r -U3 lokalize-20.08.0.orig/src/glossary/glossaryview.cpp lokalize-20.08.0/src/glossary/glossaryview.cpp
--- lokalize-20.08.0.orig/src/glossary/glossaryview.cpp	2020-08-17 15:27:13.375356507 +0300
+++ lokalize-20.08.0/src/glossary/glossaryview.cpp	2020-08-17 15:27:54.740504428 +0300
@@ -166,7 +166,7 @@
 
     bool found = false;
     //m_flowLayout->setEnabled(false);
-    foreach (const QByteArray& termId, QSet<QByteArray>(termIds.begin(), termIds.end())) {
+    foreach (const QByteArray& termId, QSet<QByteArray>::fromList(termIds)) {
         // now check which of them are really hits...
         foreach (const QString& enTerm, glossary.terms(termId, sourceLangCode)) {
             // ...and if so, which part of termEn list we must thank for match ...
