diff --git a/src/qmlcontrols/kcmcontrols/qml/private/GridViewInternal.qml b/src/qmlcontrols/kcmcontrols/qml/private/GridViewInternal.qml
index b55dea3..5d09a3d 100644
--- a/src/qmlcontrols/kcmcontrols/qml/private/GridViewInternal.qml
+++ b/src/qmlcontrols/kcmcontrols/qml/private/GridViewInternal.qml
@@ -37,7 +37,7 @@ GridView {
     anchors {
         fill: parent
         margins: 2
-        leftMargin: scroll.QtControls.ScrollBar.vertical.visible ? 2 :  internal.scrollBarSpace/2 + 2
+        leftMargin: scroll.QtControls.ScrollBar.vertical.visible ? 2 : Math.round(internal.scrollBarSpace/2) + 2
     }
     clip: true
     activeFocusOnTab: true
