diff -r -U3 kquickcharts-5.71.0.orig/src/shaders/sdf.glsl kquickcharts-5.71.0/src/shaders/sdf.glsl
--- kquickcharts-5.71.0.orig/src/shaders/sdf.glsl	2020-06-06 22:35:38.000000000 +0300
+++ kquickcharts-5.71.0/src/shaders/sdf.glsl	2020-06-18 17:56:16.088128463 +0300
@@ -14,7 +14,7 @@
 #define SDF_POLYGON_MAX_POINT_COUNT 400
 
 // A constant for pi
-const lowp float pi = acos(-1.0);
+const lowp float pi = 3.1415926535897932384626433832795;
 
 /*********************************
     Shapes
