diff -up plasma-workspace-5.12.5/startkde/startkde.cmake.startkde plasma-workspace-5.12.5/startkde/startkde.cmake
--- plasma-workspace-5.12.5/startkde/startkde.cmake.startkde    2018-05-01 08:03:40.000000000 -0500
+++ plasma-workspace-5.12.5/startkde/startkde.cmake     2018-05-06 21:12:49.592504191 -0500
@@ -279,22 +237,21 @@ if test $? -ne 0; then
   # Startup error
   echo 'startkde: Could not sync environment to dbus.'  1>&2
   test -n "$ksplash_pid" && kill "$ksplash_pid" 2>/dev/null
-  xmessage -geometry 500x100 "Could not sync environment to dbus."
+  xmessage "Could not sync environment to dbus."
   exit 1
 fi
 
-# We set LD_BIND_NOW to increase the efficiency of kdeinit.
-# kdeinit unsets this variable before loading applications.
-LD_BIND_NOW=true @CMAKE_INSTALL_FULL_LIBEXECDIR_KF5@/start_kdeinit_wrapper --kded +kcminit_startup
+@CMAKE_INSTALL_FULL_LIBEXECDIR_KF5@/start_kdeinit_wrapper --kded +kcminit_startup
 if test $? -ne 0; then
   # Startup error
   echo 'startkde: Could not start kdeinit5. Check your installation.'  1>&2
   test -n "$ksplash_pid" && kill "$ksplash_pid" 2>/dev/null
-  xmessage -geometry 500x100 "Could not start kdeinit5. Check your installation."
+  xmessage "Could not start kdeinit5. Check your installation."
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
+qdbus=qdbus-qt5
+$qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
@@ -327,13 +284,13 @@ if test x"$wait_drkonqi"x = x"true"x ; t
     # wait for remaining drkonqi instances with timeout (in seconds)
     wait_drkonqi_timeout=`kreadconfig5 --file startkderc --group WaitForDrKonqi --key Timeout --default 900`
     wait_drkonqi_counter=0
-    while qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
+    while $qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
         sleep 5
         wait_drkonqi_counter=$((wait_drkonqi_counter+5))
         if test "$wait_drkonqi_counter" -ge "$wait_drkonqi_timeout" ; then
             # ask remaining drkonqis to die in a graceful way
-            qdbus | grep 'org.kde.drkonqi-' | while read address ; do
-                qdbus "$address" "/MainApplication" "quit"
+            $qdbus | grep 'org.kde.drkonqi-' | while read address ; do
+                $qdbus "$address" "/MainApplication" "quit"
             done
             break
         fi
diff -up plasma-workspace-5.12.5/startkde/startplasma.cmake.startkde plasma-workspace-5.12.5/startkde/startplasma.cmake
--- plasma-workspace-5.12.5/startkde/startplasma.cmake.startkde 2018-05-01 08:03:40.000000000 -0500
+++ plasma-workspace-5.12.5/startkde/startplasma.cmake  2018-05-06 21:11:54.749023404 -0500
@@ -140,7 +140,8 @@ if test $? -ne 0; then
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
+qdbus=qdbus-qt5
+$qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
@@ -169,13 +170,13 @@ if test x"$wait_drkonqi"x = x"true"x ; t
     # wait for remaining drkonqi instances with timeout (in seconds)
     wait_drkonqi_timeout=`kreadconfig5 --file startkderc --group WaitForDrKonqi --key Timeout --default 900`
     wait_drkonqi_counter=0
-    while qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
+    while $qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
         sleep 5
         wait_drkonqi_counter=$((wait_drkonqi_counter+5))
         if test "$wait_drkonqi_counter" -ge "$wait_drkonqi_timeout" ; then
             # ask remaining drkonqis to die in a graceful way
-            qdbus | grep 'org.kde.drkonqi-' | while read address ; do
-                qdbus "$address" "/MainApplication" "quit"
+            $qdbus | grep 'org.kde.drkonqi-' | while read address ; do
+                $qdbus "$address" "/MainApplication" "quit"
             done
             break
         fi
diff -up plasma-workspace-5.12.5/startkde/startplasmacompositor.cmake.startkde plasma-workspace-5.12.5/startkde/startplasmacompositor.cmake
--- plasma-workspace-5.12.5/startkde/startplasmacompositor.cmake.startkde       2018-05-01 08:03:40.000000000 -0500
+++ plasma-workspace-5.12.5/startkde/startplasmacompositor.cmake        2018-05-06 21:11:54.749023404 -0500
@@ -3,6 +3,8 @@
 #  DEFAULT Plasma STARTUP SCRIPT ( @PROJECT_VERSION@ )
 #

+qdbus=qdbus-qt5
+
 # We need to create config folder so we can write startupconfigkeys
 if [  ${XDG_CONFIG_HOME} ]; then
   configDir=$XDG_CONFIG_HOME;
@@ -120,12 +122,12 @@ fi

 # Get a property value from org.freedesktop.locale1
 queryLocale1() {
-    qdbus --system org.freedesktop.locale1 /org/freedesktop/locale1 "$1"
+    $qdbus --system org.freedesktop.locale1 /org/freedesktop/locale1 "$1"
 }

 # Query whether org.freedesktop.locale1 is available. If it is, try to
 # set XKB_DEFAULT_{MODEL,LAYOUT,VARIANT,OPTIONS} accordingly.
-if qdbus --system org.freedesktop.locale1 >/dev/null 2>/dev/null; then
+if $qdbus --system org.freedesktop.locale1 >/dev/null 2>/dev/null; then
     # Do not overwrite existing values. There is no point in setting only some
     # of them as then they would not match anymore.
     if [ -z "${XKB_DEFAULT_MODEL}" -a -z "${XKB_DEFAULT_LAYOUT}" -a \
@@ -175,7 +177,7 @@ fi
 export XDG_DATA_DIRS

 # Make sure that D-Bus is running
-if qdbus >/dev/null 2>/dev/null; then
+if $qdbus >/dev/null 2>/dev/null; then
     : # ok
 else
     echo 'startplasmacompositor: Could not start D-Bus. Can you call qdbus?'  1>&2
