diff -r -U3 elisa-20.08.0.orig/src/mediaplaylistproxymodel.cpp elisa-20.08.0/src/mediaplaylistproxymodel.cpp
--- elisa-20.08.0.orig/src/mediaplaylistproxymodel.cpp	2020-08-05 11:30:36.000000000 +0300
+++ elisa-20.08.0/src/mediaplaylistproxymodel.cpp	2020-08-14 15:00:50.686352914 +0300
@@ -708,7 +708,7 @@
     for (int i = 0; i < d->mLoadPlaylist.mediaCount(); ++i) {
 #if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
         newTracks.push_back({{{{DataTypes::ElementTypeRole, ElisaUtils::FileName},
-                               {DataTypes::ResourceRole, d->mLoadPlaylist.media(i).request().url()}}}, {}, {}});
+                               {DataTypes::ResourceRole, d->mLoadPlaylist.media(i).canonicalUrl()}}}, {}, {}});
 #else
         newTracks.push_back({{{{DataTypes::ElementTypeRole, ElisaUtils::FileName},
                                {DataTypes::ResourceRole, d->mLoadPlaylist.media(i).request().url()}}}, {}, {}});
