diff -r -U3 khotkeys-5.19.0.orig/kcm_hotkeys/actions/dbus_action_widget.cpp khotkeys-5.19.0/kcm_hotkeys/actions/dbus_action_widget.cpp
--- khotkeys-5.19.0.orig/kcm_hotkeys/actions/dbus_action_widget.cpp	2020-06-04 14:10:37.000000000 +0300
+++ khotkeys-5.19.0/kcm_hotkeys/actions/dbus_action_widget.cpp	2020-06-16 13:37:04.604864340 +0300
@@ -117,7 +117,7 @@
 
 void DbusActionWidget::launchDbusBrowser() const
     {
-        auto *job = new KIO::CommandLauncherJob("qdbusviewer");
+        auto *job = new KIO::CommandLauncherJob("qdbusviewer-qt5");
         job->setUiDelegate(new KDialogJobUiDelegate(KJobUiDelegate::AutoHandlingEnabled, window()));
         job->start();
     }
