diff -r -U3 kpmcore-20.12.3.orig/src/core/smartparser.cpp kpmcore-20.12.3/src/core/smartparser.cpp
--- kpmcore-20.12.3.orig/src/core/smartparser.cpp	2021-02-25 02:06:58.000000000 +0300
+++ kpmcore-20.12.3/src/core/smartparser.cpp	2021-03-17 14:30:49.192387358 +0300
@@ -117,7 +117,7 @@
     if (m_SmartOutput.isEmpty()) {
         ExternalCommand smartctl(QStringLiteral("smartctl"), { QStringLiteral("--all"), QStringLiteral("--json"), devicePath() });
 
-        if (smartctl.run() && smartctl.exitCode() == 0) {
+        if (smartctl.run() && (smartctl.exitCode() & 1) == 0 && (smartctl.exitCode() & 2) == 0) {
             QByteArray output = smartctl.rawOutput();
 
             m_SmartOutput = QJsonDocument::fromJson(output);
