diff --git a/org.kde.kalendar.desktop b/org.kde.kalendar.desktop
--- a/org.kde.kalendar.desktop
+++ b/org.kde.kalendar.desktop
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: CC0-1.0
 
 [Desktop Entry]
+Version=1.0
 Name=Kalendar
 Name[az]=Kalendar
 Name[ca]=Kalendar
@@ -72,4 +73,4 @@ Type=Application
 Terminal=false
 StartupNotify=true
 MimeType=text/calendar;
-SingleMainWindow=true
+X-SingleMainWindow=true
