diff --git a/src/contents/ui/IncidenceInfo.qml b/src/contents/ui/IncidenceInfo.qml
index acb4564..1fcffaf 100644
--- a/src/contents/ui/IncidenceInfo.qml
+++ b/src/contents/ui/IncidenceInfo.qml
@@ -158,7 +158,7 @@ Kirigami.OverlayDrawer {
                         Kirigami.Heading {
                             Layout.fillWidth: true
 
-                            text: "<b>" + incidenceInfo.incidenceWrapper.summary + "</b>"
+                            text: "<b>" + incidenceInfo.incidenceWrapper.summaryEscaped + "</b>"
                             wrapMode: Text.Wrap
                         }
                         Kirigami.Icon {
diff --git a/src/incidencewrapper.cpp b/src/incidencewrapper.cpp
index 11a5635..652f844 100644
--- a/src/incidencewrapper.cpp
+++ b/src/incidencewrapper.cpp
@@ -155,6 +155,11 @@ QString IncidenceWrapper::summary() const
     return m_incidence->summary();
 }
 
+QString IncidenceWrapper::summaryEscaped() const
+{
+    return m_incidence->summary().toHtmlEscaped();
+}
+
 void IncidenceWrapper::setSummary(const QString &summary)
 {
     m_incidence->setSummary(summary);
diff --git a/src/incidencewrapper.h b/src/incidencewrapper.h
index f1b2ad9..fe3fe3c 100644
--- a/src/incidencewrapper.h
+++ b/src/incidencewrapper.h
@@ -41,6 +41,7 @@ class IncidenceWrapper : public QObject, public Akonadi::ItemMonitor
     Q_PROPERTY(qint64 collectionId READ collectionId WRITE setCollectionId NOTIFY collectionIdChanged)
     Q_PROPERTY(QString parent READ parent WRITE setParent NOTIFY parentChanged)
     Q_PROPERTY(QString summary READ summary WRITE setSummary NOTIFY summaryChanged)
+    Q_PROPERTY(QString summaryEscaped READ summaryEscaped NOTIFY summaryChanged)
     Q_PROPERTY(QStringList categories READ categories WRITE setCategories NOTIFY categoriesChanged)
     Q_PROPERTY(QString description READ description WRITE setDescription NOTIFY descriptionChanged)
     Q_PROPERTY(QString location READ location WRITE setLocation NOTIFY locationChanged)
@@ -117,6 +118,7 @@ public:
     QString parent() const;
     void setParent(QString parent);
     QString summary() const;
+    QString summaryEscaped() const;
     void setSummary(const QString &summary);
     QStringList categories();
     void setCategories(QStringList categories);
