diff -up kwordquiz-18.12.3/src/kwordquiz.cpp.entries kwordquiz-18.12.3/src/kwordquiz.cpp
--- kwordquiz-18.12.3/src/kwordquiz.cpp.entries	2019-01-28 00:10:43.000000000 -0600
+++ kwordquiz-18.12.3/src/kwordquiz.cpp	2019-05-10 15:41:24.149705647 -0500
@@ -485,7 +485,7 @@ void KWordQuizApp::initDocument()
   m_doc->identifier(1).setName(i18n("Column 2"));
   // m_doc->identifier(1).setLocale("en");
 
-  for (int i=0; i<20; i++)
+  for (int i=0; i<100; i++)
   {
     m_doc->lesson()->appendEntry(new KEduVocExpression());
   }
