diff -up baloo-5.45.0/src/file/fileindexerconfig.cpp.baloofile_config baloo-5.45.0/src/file/fileindexerconfig.cpp
--- baloo-5.45.0/src/file/fileindexerconfig.cpp.baloofile_config	2018-04-07 14:32:21.000000000 -0500
+++ baloo-5.45.0/src/file/fileindexerconfig.cpp	2018-04-08 07:53:43.984930379 -0500
@@ -318,8 +318,13 @@ void FileIndexerConfig::buildFolderCache
     }
 
     KConfigGroup group = m_config.group("General");
-    QStringList includeFoldersPlain = group.readPathEntry("folders", QStringList() << QDir::homePath());
-    QStringList excludeFoldersPlain = group.readPathEntry("exclude folders", QStringList());
+    QStringList includeFoldersPlain = group.readPathEntry("folders", QStringList() << 
+        QStandardPaths::standardLocations(QStandardPaths::DocumentsLocation) <<
+        QStandardPaths::standardLocations(QStandardPaths::MusicLocation) <<
+        QStandardPaths::standardLocations(QStandardPaths::MoviesLocation) <<
+        QStandardPaths::standardLocations(QStandardPaths::PicturesLocation) 
+    );
+    QStringList excludeFoldersPlain = group.readPathEntry("exclude folders", QStringList() << QDir::homePath());
 
     // Add all removable media and network shares as ignored unless they have
     // been explicitly added in the include list
