diff --git a/kcms/nightcolor/package/contents/ui/TimeField.qml b/kcms/nightcolor/package/contents/ui/TimeField.qml
--- a/kcms/nightcolor/package/contents/ui/TimeField.qml
+++ b/kcms/nightcolor/package/contents/ui/TimeField.qml
@@ -57,8 +57,7 @@
         var minutes = text.slice(3, 5);
 
         var date = new Date();
-        date.setHours(hours);
-        date.setMinutes(minutes);
+        date.setHours(hours, minutes, 0, 0);
 
         backend = date;
     }

