diff -r -U3 plasma-desktop-5.19.2.orig/desktoppackage/contents/views/Desktop.qml plasma-desktop-5.19.2/desktoppackage/contents/views/Desktop.qml
--- plasma-desktop-5.19.2.orig/desktoppackage/contents/views/Desktop.qml	2020-06-23 15:15:32.000000000 +0300
+++ plasma-desktop-5.19.2/desktoppackage/contents/views/Desktop.qml	2020-06-30 20:33:19.420140406 +0300
@@ -162,7 +162,7 @@
                             ActivitySwitcher.Backend.shouldShowSwitcher
                         sidePanel.hideOnWindowDeactivate = Qt.binding(function() {
                             return !ActivitySwitcher.Backend.shouldShowSwitcher
-                                && !sidePanelStack.item.showingDialog;
+                                && (sidePanelStack.item && !sidePanelStack.item.showingDialog);
                         })
                         sidePanelStack.item.forceActiveFocus();
                     } else if (sidePanelStack.state == "widgetExplorer"){
