diff -up extra-cmake-modules-5.89.0/kde-modules/KDEInstallDirs5.cmake.qt_prefix extra-cmake-modules-5.89.0/kde-modules/KDEInstallDirs5.cmake
--- extra-cmake-modules-5.89.0/kde-modules/KDEInstallDirs5.cmake.qt_prefix	2021-12-02 00:14:53.000000000 -0600
+++ extra-cmake-modules-5.89.0/kde-modules/KDEInstallDirs5.cmake	2021-12-08 09:23:47.194443267 -0600
@@ -236,7 +236,7 @@ include("${ECM_MODULE_DIR}/ECMQueryQmake
 set(_default_KDE_INSTALL_USE_QT_SYS_PATHS OFF)
 if(NOT DEFINED KDE_INSTALL_USE_QT_SYS_PATHS)
     query_qmake(qt_install_prefix_dir QT_INSTALL_PREFIX TRY)
-    if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}")
+    if( (qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}") OR ( ("${CMAKE_INSTALL_PREFIX}" STREQUAL "/usr") AND (qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_FULL_LIBDIR}/qt5")) )
         message(STATUS "Installing in the same prefix as Qt, adopting their path scheme.")
         set(_default_KDE_INSTALL_USE_QT_SYS_PATHS ON)
     endif()
diff -up extra-cmake-modules-5.89.0/kde-modules/KDEInstallDirs6.cmake.qt_prefix extra-cmake-modules-5.89.0/kde-modules/KDEInstallDirs6.cmake
