diff --git a/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp b/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
index c796aa2..119651b 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
+++ b/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
@@ -394,7 +394,7 @@ void PackageKitBackend::checkForUpdates()
 
     if (!m_refresher) {
         acquireFetching(true);
-        m_refresher = PackageKit::Daemon::refreshCache(false);
+        m_refresher = PackageKit::Daemon::refreshCache(true);
 
         connect(m_refresher.data(), &PackageKit::Transaction::errorCode, this, &PackageKitBackend::transactionError);
         connect(m_refresher.data(), &PackageKit::Transaction::finished, this, [this]() {
diff --git a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
index cba6b72..5bcd3b7 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
+++ b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
@@ -51,7 +51,7 @@ PackageKitNotifier::PackageKitNotifier(QObject *parent)
             bool ok;
             const int days = value.toInt(&ok);
             if (!ok || days == 0) {
-                regularCheck->setInterval(24 * 60 * 60 * 1000); // refresh at least once every day
+                regularCheck->setInterval(12 * 60 * 60 * 1000); //refresh at least twice every day
                 regularCheck->start();
                 if (!value.isEmpty())
                     qWarning() << "couldn't understand value for timer:" << value;
@@ -229,7 +229,7 @@ void PackageKitNotifier::onDistroUpgrade(PackageKit::Transaction::DistroUpgrade
 void PackageKitNotifier::refreshDatabase()
 {
     if (!m_refresher) {
-        m_refresher = PackageKit::Daemon::refreshCache(false);
+        m_refresher = PackageKit::Daemon::refreshCache(true);
         connect(m_refresher.data(), &PackageKit::Transaction::finished, this, &PackageKitNotifier::recheckSystemUpdateNeeded);
     }
 
