diff -r -U3 kde-dev-scripts-19.04.0.orig/kde-systemsettings-tree.py kde-dev-scripts-19.04.0/kde-systemsettings-tree.py
--- kde-dev-scripts-19.04.0.orig/kde-systemsettings-tree.py	2019-03-16 20:45:58.000000000 +0300
+++ kde-dev-scripts-19.04.0/kde-systemsettings-tree.py	2019-05-07 15:30:03.557033414 +0300
@@ -39,26 +39,26 @@
 def printLevel(category, depth=0):
 	results = KServiceTypeTrader.self().query('KCModule', "([X-KDE-System-Settings-Parent-Category] == '%s')"%(category))
 	for result in results:
-		print ("	"*depth)+'%s [%s]' % (result.property('Name', QVariant.String).toString(), result.desktopEntryPath())
+		print(("	"*depth)+'%s [%s]' % (result.property('Name', QVariant.String).toString(), result.desktopEntryPath()))
 		if result.property('Comment', QVariant.String).toString().isEmpty():
-			print("	"*depth)+"***No comment!***"
+			print(("	"*depth)+"***No comment!***")
 	categories = KServiceTypeTrader.self().query('SystemSettingsCategory', "([X-KDE-System-Settings-Parent-Category] == '%s')"%(category))
 	for subcategory in categories:
-		print ("	"*depth)+'%s [%s]' % (subcategory.property('Name', QVariant.String).toString(), subcategory.desktopEntryPath())
+		print(("	"*depth)+'%s [%s]' % (subcategory.property('Name', QVariant.String).toString(), subcategory.desktopEntryPath()))
 		printLevel(subcategory.property('X-KDE-System-Settings-Category', QVariant.String).toString(), depth+1)
 
-print "*** Current organization:"
+print("*** Current organization:")
 printLevel('')
 
-print "\n"
-print "*** Searching for invalid categories:"
-print "**** KCModules:"
+print("\n")
+print("*** Searching for invalid categories:")
+print("**** KCModules:")
 results = KServiceTypeTrader.self().query('KCModule')
 for result in results:
 	if (not (result.property('X-KDE-System-Settings-Parent-Category', QVariant.String).toString() in VALID_CATEGORIES)):
-		print "%s [%s]: %s"%(result.property('Name', QVariant.String).toString(), result.desktopEntryPath(), result.property('X-KDE-System-Settings-Parent-Category', QVariant.String).toString())
-print "**** Categories:" 
+		print("%s [%s]: %s"%(result.property('Name', QVariant.String).toString(), result.desktopEntryPath(), result.property('X-KDE-System-Settings-Parent-Category', QVariant.String).toString()))
+print("**** Categories:")
 results = KServiceTypeTrader.self().query('SystemSettingsCategory')
 for result in results:
 	if (not (result.property('X-KDE-System-Settings-Parent-Category', QVariant.String).toString() in VALID_CATEGORIES)):
-		print "%s [%s]: %s"%(result.property('Name', QVariant.String).toString(), result.desktopEntryPath(), result.property('X-KDE-System-Settings-Parent-Category', QVariant.String).toString())
+		print("%s [%s]: %s"%(result.property('Name', QVariant.String).toString(), result.desktopEntryPath(), result.property('X-KDE-System-Settings-Parent-Category', QVariant.String).toString()))
