diff --git a/kcms/kfontinst/lib/FcEngine.cpp b/kcms/kfontinst/lib/FcEngine.cpp
index 36ecf65c5909c94b29f04d74bffcaf66723db9ca..74b73cbfff59481145c5b441ad9225250342b4a2 100644
--- a/kcms/kfontinst/lib/FcEngine.cpp
+++ b/kcms/kfontinst/lib/FcEngine.cpp
@@ -576,6 +576,14 @@ QImage CFcEngine::Xft::toImage(int w, int h) const
     if (!xImage) {
         return QImage();
     }
+    if (xImage->depth == 24) {
+        // the RGB32 format requires data format 0xffRRGGBB, ensure that this fourth byte really is 0xff
+        // (i.e. when using NVidia proprietary driver)
+        auto lData = reinterpret_cast<quint32 *>(xImage->data);
+        for (size_t iIter = 0; iIter < xImage->width * xImage->height; iIter++) {
+            lData[iIter] |= 0xff000000;
+        }
+    }
     return QImage(xImage->data, xImage->width, xImage->height, xImage->stride, imageFormat, &cleanupXImage, xImage);
 }
     
