diff -r -U3 plasma-workspace-5.20.0.orig/containmentactions/contextmenu/menu.cpp plasma-workspace-5.20.0/containmentactions/contextmenu/menu.cpp
--- plasma-workspace-5.20.0.orig/containmentactions/contextmenu/menu.cpp	2020-10-09 15:31:38.000000000 +0300
+++ plasma-workspace-5.20.0/containmentactions/contextmenu/menu.cpp	2020-10-13 23:06:47.443332356 +0300
@@ -30,6 +30,7 @@
 #include <QIcon>
 #include <KGlobalAccel>
 #include <KLocalizedString>
+#include <KToolInvocation>
 
 #include <Plasma/Containment>
 #include <Plasma/Corona>
@@ -40,6 +41,7 @@
 
 ContextMenu::ContextMenu(QObject *parent, const QVariantList &args)
     : Plasma::ContainmentActions(parent, args),
+      m_runKonsoleAction(nullptr),
       m_runCommandAction(nullptr),
       m_lockScreenAction(nullptr),
       m_logoutAction(nullptr),
@@ -69,6 +71,7 @@
         c->containmentType() == Plasma::Types::CustomPanelContainment) {
         m_actionOrder   << QStringLiteral("add widgets")
                         << QStringLiteral("_add panel")
+                        << QStringLiteral("_run_konsole")
                         << QStringLiteral("_context")
                         << QStringLiteral("configure")
                         << QStringLiteral("remove");
@@ -105,6 +108,10 @@
 
     // everything below should only happen once, so check for it
     if (!m_runCommandAction) {
+        m_runKonsoleAction = new QAction(i18n("Konsole"), this);
+        m_runKonsoleAction->setIcon(QIcon::fromTheme("utilities-terminal"));
+        connect(m_runKonsoleAction, &QAction::triggered, this, &ContextMenu::runKonsole);
+
         m_runCommandAction = new QAction(i18nc("plasma_containmentactions_contextmenu", "Show KRunner"), this);
         m_runCommandAction->setIcon(QIcon::fromTheme(QStringLiteral("plasma-search")));
         m_runCommandAction->setShortcut(KGlobalAccel::self()->globalShortcut(QStringLiteral("krunner.desktop"), QStringLiteral("_launch")).value(0));
@@ -185,6 +192,8 @@
         if (c->corona() && c->corona()->immutability() == Plasma::Types::Mutable) {
             return c->corona()->actions()->action(QStringLiteral("add panel"));
         }
+    } else if (name == QLatin1String("_run_konsole")) {
+        return m_runKonsoleAction;
     } else if (name == QLatin1String("_run_command")) {
         if (KAuthorized::authorizeAction(QStringLiteral("run_command")) && KAuthorized::authorize(QStringLiteral("run_command"))) {
             return m_runCommandAction;
@@ -212,6 +221,11 @@
     return nullptr;
 }
 
+void ContextMenu::runKonsole()
+{
+    KToolInvocation::invokeTerminal(QString(), QDir::homePath());
+}
+
 void ContextMenu::runCommand()
 {
     if (!KAuthorized::authorizeAction(QStringLiteral("run_command"))) {
Только в plasma-workspace-5.20.0/containmentactions/contextmenu: menu.cpp.orig
Только в plasma-workspace-5.20.0/containmentactions/contextmenu: menu.cpp.rej
diff -r -U3 plasma-workspace-5.20.0.orig/containmentactions/contextmenu/menu.h plasma-workspace-5.20.0/containmentactions/contextmenu/menu.h
--- plasma-workspace-5.20.0.orig/containmentactions/contextmenu/menu.h	2020-10-09 15:31:38.000000000 +0300
+++ plasma-workspace-5.20.0/containmentactions/contextmenu/menu.h	2020-10-13 23:00:15.017991124 +0300
@@ -42,10 +42,12 @@
     void save(KConfigGroup &config) override;
 
 public Q_SLOTS:
+    void runKonsole();
     void runCommand();
     void startLogout();
 
 private:
+    QAction *m_runKonsoleAction;
     QAction *m_runCommandAction;
     QAction *m_lockScreenAction;
     QAction *m_logoutAction;
