diff -r -U3 plasma-workspace-5.20.2.orig/lookandfeel/contents/splash/Splash.qml plasma-workspace-5.20.2/lookandfeel/contents/splash/Splash.qml
--- plasma-workspace-5.20.2.orig/lookandfeel/contents/splash/Splash.qml	2020-10-27 15:29:48.000000000 +0300
+++ plasma-workspace-5.20.2/lookandfeel/contents/splash/Splash.qml	2020-11-09 23:40:09.569394269 +0300
@@ -40,7 +40,7 @@
     Item {
         id: content
         anchors.fill: parent
-        opacity: 0
+        //opacity: 0
         TextMetrics {
             id: units
             text: "M"
