From c65264e15355397931df8ad7fa58619b26487c17 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Wed, 26 Oct 2022 21:43:41 +0200
Subject: [PATCH] x11window: revert more from 3a28c02f

BUG: 461032
---
 src/x11window.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/x11window.cpp b/src/x11window.cpp
index eb0e66152b..8783375df7 100644
--- a/src/x11window.cpp
+++ b/src/x11window.cpp
@@ -4509,11 +4509,10 @@ void X11Window::maximize(MaximizeMode mode)
                 }
             }
             r.moveTopLeft(rules()->checkPosition(r.topLeft()));
-            // The above code tries to center align the window followed by setting top and bottom
-            // it's possible that we're in between two pixels
-            r.setX(Xcb::nativeFloor(r.x()));
-            r.setY(Xcb::nativeFloor(r.y()));
         }
+        // The above code tries to center align the window followed by setting top and bottom
+        // it's possible that we're in between two pixels
+        r = Xcb::nativeFloor(r);
 
         moveResize(r);
         if (options->electricBorderMaximize() && r.top() == clientArea.top()) {
-- 
GitLab

